src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs

```diff
@@ -84,6 +85,15 @@ public override async Task RunAsync(CancellationToken cancellationToken)
             foreach (var (name, secret, secretType, secretReferences) in orderedSecretTypes)
             {
                 _console.WriteLine($"Synchronizing secret {name}, type {secret.Type}");
+
+                if (existingSecrets.TryGetValue(name, out var existingSecret))
+                {
+                    if (!existingSecret.Tags.ContainsKey(AzureKeyVault.NextRotationOnTag))
```

Comment:
I'd merge these ifs, but it's a personal preference

Reply:
done